auth_pass_ip_restriction/models/res_users.py

```diff
@@ -0,0 +1,15 @@
+from odoo import _, models
+from odoo.exceptions import AccessDenied
+from odoo.http import request
+from odoo.tools.config import config
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    def _check_credentials(self, password, env):
```

Comment:
In our particular usecase there are some authentication methods which should be able to bypass this... so we should build in here that if `env` contains `bypass_ip_check: True` then we don't do the below check

Reply:
Added it.